### DISCUSSION TITLE
how to make the task(sub workflow task) delay to run in Dynamic Workflows?

### DISCUSSION THREAD

**Discussion by andyyumiao** [Netflix/conductor#2518] [Q&A]
As title, in latest conductor version, how to make the task(sub workflow task) delay to run in Dynamic Workflows?For example, I want a sub workflow task run after 60s when the task starts to be scheduling, .

**Comment by aravindanr** [comment]
When defining the workflow, you can use the `startDelay` parameter in the task to delay the start of a task. The start delay is specified in seconds.

**Reply by andyyumiao** [reply]
@aravindanr hi,  the startDelay parameter works in defined workflow, but doesn't work in dynamic workflow(i mean one-time workflow)

**Reply by jxu-nflx** [reply]
Hello @andyyumiao `startDelay` param should also work for dynamic workflow. For an example, you could add `startDelay` in this sample dynamic workflow.
```
{
  "name": "my_adhoc_unregistered_workflow",
  "workflowDef": {
    "ownerApp": "my_owner_app",
    "ownerEmail": "my_owner_email@test.com",
    "createdBy": "my_username",
    "name": "my_adhoc_unregistered_workflow",
    "description": "Test Workflow setup",
    "version": 1,
    "tasks": [
        {
            "name": "fetch_data",
            "type": "HTTP",
            "taskReferenceName": "fetch_data",
           "startDelay":60,
            "inputParameters": {
              "http_request": {
                "connectionTimeOut": "3600",
                "readTimeOut": "3600",
                "uri": "${workflow.input.uri}",
                "method": "GET",
                "accept": "application/json",
                "content-Type": "application/json",
                "headers": {
                }
              }
            },
            "taskDefinition": {
                "name": "fetch_data",
                "retryCount": 0,
                "timeoutSeconds": 3600,
                "timeoutPolicy": "TIME_OUT_WF",
                "retryLogic": "FIXED",
                "retryDelaySeconds": 0,
                "responseTimeoutSeconds": 3000
            }
        }
    ],
    "outputParameters": {
    }
  },
  "input": {
    "uri": "http://www.google.com"
  }
}
```

**Reply by andyyumiao** [reply]
>  "startDelay":60,

@jxu-nflx it's works fine in http task in dynamic workflow, but it doesn't work for SUB_WORFLOW task in dynamic workflow.
My steps are like this：

1、create a general workflow：
{
  "createTime": 1637048805478,
  "updateTime": 1618812084028,
  "name": "19test009",
  "description": "test workflow",
  "version": 1,
  "tasks": [
    {
      "name": "fetch_data",
      "taskReferenceName": "fetch_data",
      "inputParameters": {
        "http_request": {
          "connectionTimeOut": "3600",
          "readTimeOut": "3600",
          "uri": "${workflow.input.uri}",
          "method": "GET",
          "accept": "application/json",
          "content-Type": "application/json",
          "headers": {}
        }
      },
      "type": "HTTP",
      "decisionCases": {},
      "defaultCase": [],
      "forkTasks": [],
      "startDelay": 0,
      "joinOn": [],
      "optional": false,
      "taskDefinition": {
        "name": "fetch_data",
        "retryCount": 0,
        "timeoutSeconds": 3600,
        "inputKeys": [],
        "outputKeys": [],
        "timeoutPolicy": "TIME_OUT_WF",
        "retryLogic": "FIXED",
        "retryDelaySeconds": 0,
        "responseTimeoutSeconds": 3000,
        "inputTemplate": {},
        "rateLimitPerFrequency": 0,
        "rateLimitFrequencyInSeconds": 1
      },
      "defaultExclusiveJoinTask": [],
      "asyncComplete": false,
      "loopOver": []
    }
  ],
  "inputParameters": [],
  "outputParameters": {},
  "schemaVersion": 2,
  "restartable": true,
  "workflowStatusListenerEnabled": true,
  "ownerEmail": "test@163.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 0,
  "variables": {},
  "inputTemplate": {}
}

2、Then run the dynamic workflow：
{
  "name": "my_adhoc_unregistered_workflow22",
  "workflowDef": {
    "ownerApp": "my_owner_app",
    "ownerEmail": "my_owner_email@test.com",
    "createdBy": "my_username",
    "name": "my_adhoc_unregistered_workflow22",
    "description": "Test Workflow setup",
    "version": 1,
    "tasks": [
        {
            "name": "19test009_abc",
            "taskReferenceName": "sub1",
            "type": "SUB_WORKFLOW",
            "startDelay":60,
            "subWorkflowParam": {
                    "name": "19test009",
                    "version": 1,
                    "taskToDomain": {
                        "*": "mydomain"
                    }
                },
            "inputParameters": {
            }
        }
    ],
    "outputParameters": {
    }
  },
  "input": {
    "uri": "http://www.google.com"
  }
}

3、the workflow '19test009' doesn't delay 60s as expected

**Comment by jxu-nflx** [comment]
Hello @andyyumiao I seem to be able to reproduce it. I'll need to investigate more. Feel free to submit a PR for it you are able to figure out a fix for it

**Reply by jxu-nflx** [reply]
https://github.com/Netflix/conductor/pull/2608

**Reply by andyyumiao** [reply]
@jxu-nflx very thankful for the fix
